src/sdk/apex-extend.ts

```diff
@@ -275,119 +206,19 @@ Apex.prototype.scan = async function ({
     }
   }
 
-  if (accessPolicy) {
-    if (typeof accessPolicy === "string") {
-      request.accessPolicy = accessPolicy;
-    }
-  }
-
-  if (contentPolicy) {
-    if (typeof contentPolicy === "string") {
-      request.contentPolicy = contentPolicy;
-    }
-  }
-
-  return this.scanRequest(request);
-};
-
-Apex.prototype.police = async function ({
-  messages,
-  files,
-  requestType,
-  annotations,
-  bypassHash,
-  anonymization,
-  provider,
-  user,
-}: {
-  messages?: string | string[] | undefined,
-  files?: string | string[] | undefined,
-  requestType?: components.Type | string | undefined,
-  annotations?: { [k: string]: string } | undefined,
-  bypassHash?: string | undefined,
-  anonymization?: components.Anonymization | string | undefined,
-  provider?: string | undefined,
-  user?: components.Policeexternaluser | undefined,
-}) {
-  const request: components.Policerequest = {};
-
-  if (messages) {
-    if (typeof messages === "string") {
-      request.messages = [messages];
-    } else if (Array.isArray(messages) && messages.every((m) => typeof m === "string")) {
-      request.messages = messages;
-    }
-  }
-
-  if ((!request.messages || request.messages.length === 0) && !files) {
-    throw new Error("no messages and no files provided");
-  }
 
-  if (files) {
-    let processFiles: string[] = [];
-    if (typeof files === "string") {
-      processFiles = [files];
-    } else if (Array.isArray(files) && files.every((f) => typeof f === "string")) {
-      processFiles = files;
-    }
-
-    const extractions: components.Extractionrequest[] = [];
-    for (const processFile of processFiles) {
-      const fileContent = await readFileAndBase64Encode(processFile);
-      extractions.push({
-        data: fileContent,
-      });
-    }
-    request.extractions = extractions;
+  let gconfig: GuardConfig;
+  try {
+    gconfig = guardConfig ? new GuardConfig(guardConfig) : new GuardConfig();
+  } catch (e) {
+    console.debug("Error while processing the guard config");
+    throw new Error("Cannot process the guard config");
```

Comment:
hm, I don't particularly like that we hide the underlying error here, but you might have your reasons?

if there is a good reason, then keep it this way, however, at least for the debug log, keep the exception please.

Reply:
yes, I don't have any good reasons, agree, its better to propagate.

Reply:
done